src/components/oauth_manage.jsx

```diff
@@ -0,0 +1,126 @@
+          </h2>
+          <div className="oauth-clients">
+            {this.props.clients.map((c, idx) => <Client {...c} key={idx} />)}
+          </div>
+        </Bootstrap.Col>
+        <Bootstrap.Col xs={12} md={4}><Help /></Bootstrap.Col>
+      </Bootstrap.Grid>
+    );
+  }
+}
+
+export class Edit extends React.Component {
+  render() {
+    const client = this.props.client || {};
+    const invalid = this.props.invalid;
+    const edit = !!this.props.client;
+
+    return (
+      <Bootstrap.Grid>
+        <Bootstrap.Col xs={12} md={8}>
+          <h2>
+            { edit ? 'Edit OAuth Client' : 'Create OAuth Client' }
+          </h2>
+          { invalid ? <Bootstrap.Alert bsStyle='danger'>
+                  There was an error with your registration.
+                </Bootstrap.Alert> : '' }
+          <form action="/oauth/edit" method="POST" encType={ edit ? undefined : "multipart/form-data" }>
+            { edit ? <input type="hidden" name="id" value={client.client_id} /> : '' }
+            <ValidatedInput invalid={invalid} name='name' type='text' label='Name'
+              placeholder='Client Name' value={client.name} />
+            <ValidatedInput invalid={invalid} name='website' type='text' label='Website' placeholder='http://example.com' value={client.website} />
+            <ValidatedInput invalid={invalid} name='hosts'
+              type='text' label='Hosts'
+              placeholder='Hosts' value={edit ? client.hosts.join(', ') : ''}
+              help='List of hosts, separated by commas. You will only be allowed to redirect to these hostnames.' />
+            { edit ? '' : (
+              <div>
+                <ValidatedInput invalid={invalid} name='logo' type='file' label='Logo' />
+                <ValidatedInput type='checkbox' name='secret' label='Confidential' help={
+                  'If checked, your client will be issued and must use a secret key. ' +
+                  'This should only be checked if you client can keep the key a secret. ' +
+                  'For example, mobile apps cannot maintain confidentiality, ' +
+                  'but backend web services can.'
+                } />
+              </div>
+            )}
+            <button className="btn btn-alt">Save</button>
+          </form>
+        </Bootstrap.Col>
+        <Bootstrap.Col xs={12} md={4}><Help /></Bootstrap.Col>
+      </Bootstrap.Grid>
+    );
+  }
+}
+
+class Help extends React.Component {
+  render() {
+    return (
+      <div>
+        <h2>Get Started with OAuth</h2>
+        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis condimentum metus eu lacus laoreet elementum. Morbi vehicula luctus tortor, eget gravida felis condimentum non. Morbi lobortis rhoncus tellus in ultrices. Duis id neque vitae diam elementum commodo et eget risus. Mauris vitae scelerisque urna. Ut in porttitor neque. Nunc euismod augue id velit molestie tristique. Curabitur rhoncus vestibulum quam.</p>
```

Comment:
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum sed nibh euismod, luctus eros in, dictum sem. Duis dapibus sodales urna, eget maximus ligula aliquam et. Suspendisse risus diam, scelerisque at scelerisque laoreet, ullamcorper consequat lectus. Aliquam maximus ante id arcu cursus, ac placerat metus suscipit. Cras ornare sed sem sit amet malesuada. Nullam arcu lorem, ullamcorper eu facilisis eget, bibendum vestibulum eros. Aliquam imperdiet id justo in tincidunt. Integer vestibulum risus felis, ac eleifend justo commodo ut. Nam vestibulum libero id gravida consequat. Ut tellus libero, tincidunt ut leo malesuada, sollicitudin convallis mi. Ut condimentum est nec lacus pharetra, in imperdiet nisl egestas. Nam ut lectus erat. Morbi rutrum ex in dolor rhoncus, at ultricies risus mattis.